cmd/kubelet/app/options/options.go

```diff
@@ -76,6 +76,12 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.TLSPrivateKeyFile, "tls-private-key-file", s.TLSPrivateKeyFile, "File containing x509 private key matching --tls-cert-file.")
 	fs.StringVar(&s.CertDirectory, "cert-dir", s.CertDirectory, "The directory where the TLS certs are located (by default /var/run/kubernetes). "+
 		"If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.")
+	fs.StringVar(&s.BootstrapAuthToken, "bootstrap-auth-token", s.BootstrapAuthToken, ""+
```

Comment:
> maybe we can make changes later to have a flag (e.g. --cluster in the proposal I pasted above) to override the kubeconfig for this bootstrap auth purpose.

My point is that the cluster discovery method will need the same information already contained in the kubeconfig (server host, CA bundle, and client credentials) to complete discovery securely and be able to work against private clusters.